nexus/reconfigurator/execution/src/lib.rs

```diff
@@ -46,19 +46,42 @@ pub struct RealizeArgs<'a> {
     pub datastore: &'a DataStore,
     pub resolver: &'a Resolver,
     pub blueprint: &'a Blueprint,
-    pub nexus_id: OmicronZoneUuid,
+    pub nexus_id: Option<OmicronZoneUuid>,
+    pub creator: OmicronZoneUuid,
```

Comment:
Previously, we were using `nexus_id` in three places:

- as the "creator" field for DNS records
- to assign _ourselves_ sagas and support bundles from any expunged Nexus instances

I've separated these out.  Now we use `creator` for the first one.  This must always be specified.  We use `nexus_id` for the other two.  You can leave this unspecified, in which case execution will skip these steps (because it makes no sense for this tool to assign itself sagas or support bundles).

I think `creator` could be a free-form string but I wasn't sure if some stuff might assume it's a uuid so I did what felt like the conservative thing and kept it a uuid.